content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,286 @@
+be using on Node, have all resources ready for Pod to take in use.
+
+```gRPC
+    // NodePrepareResources prepares several ResourceClaims
+    // for use on the node. If an error is returned, the
+    // response is ignored. Failures for individual claims
+    // can be reported inside NodePrepareResourcesResponse.
+    NodePrepareResources(context.Context, *NodePrepareResourcesRequest) (*NodePrepareResourcesResponse, error)
+    // NodeUnprepareResources is the opposite of NodePrepareResources.
+    // The same error handling rules apply,
+    NodeUnprepareResources(context.Context, *NodeUnprepareResourcesRequest) (*NodeUnprepareResourcesResponse, error)
+```
+
+### Example {#example-pod}
+
+Suppose a Kubernetes cluster is running a resource driver gpu.resource.example.com with Resource
+Class `example.example.com`. Here is an example of a pod requesting this resource to run a demo
+workload:
+
+```yaml
+apiVersion: gpu.resource.example.com/v1alpha1
+kind: GpuClaimParameters
+metadata:
+  name: single-gpu
+spec:
+  count: 1
+---
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClaim
+metadata:
+  name: gpu-test
+spec:
+  resourceClassName: gpu.example.com
+  parametersRef:
+    apiGroup: gpu.resource.example.com/v1alpha1
+    kind: GpuClaimParameters
+    name: single-gpu
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: gpu-test4
+  name: pod0
+  labels:
+    app: pod
+spec:
+  containers:
+  - name: container1
+    image: ubuntu:22.04
+    command: ["bash", "-c"]
+    args: ["export; sleep 9999"]
+    resources:
+      claims:
+      - name: gpus
+  resourceClaims:
+  - name: gpus
+    source:
+      resourceClaimTemplateName: gpu-test
+# This Pod wants to use ResourceClaim gpu-test that needs 1 device of ResourceClass
+# gpu.example.com, that is handled by the Resource Driver gpu.resource.example.com .
+#
+# The Resource Driver will allocate the needed resource to ResourceClaim and ensure it is
+# ready to use, only then the Pod will start.
+```
+
+## Resource Driver deployment
+
+The recommended way to deploy a resource driver is a Deployment for controller part and a DaemonSet
+for kubelet-plugin part. It is also possible to deploy it as a package for your node's
+operating system, or manually.
+
+Kubelet will be using gRPC interface to interact with Resource Driver's kubelet-plugin. No special
+permissions are required for resource-driver.
+
+ResourceClass should be part of Resource Driver deployment.
+
+## API compatibility
+
+Kubernetes Dynamic Resource Allocation support is in alpha. The API may change before stabilization,
+in incompatible ways. As a project, Kubernetes recommends that resource driver developers:
+
+* Watch for changes in future releases.
+* Support multiple versions of the resource driver API for backward/forward compatibility.
+
+If you enable the DynamicResourceAllocation feature and run resource driver kubelet-plugin on nodes
```

Comment:
```suggestion
If you enable the `DynamicResourceAllocation` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) and run associated kubelet plugins on nodes
```

Reply:
I'll add a glossary tooltip instead of plain hyperlink.

Reply:
Please use this particular formulation. We use a hyperlink not tooltips when referring to feature gates - see other docs for examples (docs for beta features will show this better than alpha docs).

Reply:
True, git grep shows 0 references to tooltip. Alright then.